src/core/handle.ml

```diff
@@ -185,7 +193,23 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss e Defin x a impl d in
+      (fst (Sig_state.add_symbol ss e Defin x a impl d), None)
+  | P_inductive(e, s, t, tl)     ->
+      (* Add the inductive type in the signature *)
+      let (ss, typ) = handle_symbol ss e Const s [] t in
+      (* Add the constructors in the signature. *)
+      let add_cons :
+                sig_state * sym list -> ident * p_term -> sig_state * sym list
+        = fun (ss, cons) (id, a) ->
+        let (ss, c) = handle_symbol ss e Injec id [] a in
+        (ss, c::cons)
+      in
+      let (ss, cons) = List.fold_left add_cons (ss, []) tl in
+      (* Compute the induction principle *)
+      let pr = Inductive.principle typ cons in
+      let ind_sym = Sign.add_symbol ss.signature e Defin
+                      ({ elt = "ind"; pos = cmd.pos }) pr [] in
+      Sign.add_inductive ss.signature typ cons ind_sym;
       (ss, None)
```

Comment:
I propose some renamings and some small changes (constructors are Const and the induction principle name):
```
  | P_inductive(e, id, a, l)     ->
      (* Add the inductive type in the signature *)
      let (ss, sym_typ) = handle_symbol ss e Const id [] a in
      (* Add the constructors in the signature. *)
      let add_cons (ss, cons_list) (id, a) ->
        let (ss, sym_cons) = handle_symbol ss e Const id [] a in
        (ss, sym_cons::cons_list)
      in
      let (ss, cons_list) = List.fold_left add_cons (ss, []) l in
      (* Compute the induction principle *)
      let ind_typ = Inductive.principle sym_typ cons_list in
      let ind_name = in_pos cmd.pos (id.elt ^ "_ind") in
      let sym_ind = Sign.add_symbol ss.signature e Defin sl ind_name [] in
      Sign.add_inductive ss.signature sym_typ cons_list sym_ind;
      (ss, None)
```

Reply:
Yes, I agree but:
* remove the signature of the function `add_cons` is maybe less readable. But as you want.
* why constructors will be constant ? One property of constructors is to be injective, no?

Reply:
Const is stronger than Injec: constant => injective.

Reply:
Ah OK perfect! :+1: